### PR TITLE
[WIP] search/commit , issue #2132

### PR DESCRIPTION
**Pull request by kentahikaru**
Initial commit, first baby steps

### CODE REVIEW

File: Octokit/Clients/SearchClient.cs [hunk 1]
```diff
@@ -83,5 +83,18 @@ public Task<SearchLabelsResult> SearchLabels(SearchLabelsRequest search)
...
+        /// search commits
+        /// https://developer.github.com/v3/search/#search-commits
+        /// </summary>
+        /// <param name="search"></param>
+        /// <returns>List of commits</returns>
+        [ManualRoute("GET", "/search/commits")]
+        public Task<SearchCommitsResult> SearchCommits(SearchLabelsRequest search)
+        {
+            Ensure.ArgumentNotNull(search, nameof(search));
+            return ApiConnection.Get<SearchLabelsResult>(ApiUrls.SearchLabels(), search.Parameters, AcceptHeaders.SearchCommitsPreview);
```
Comment (shiftkey) [inline Octokit/Clients/SearchClient.cs]:
```suggestion
            return ApiConnection.Get<SearchCommitsResult>(ApiUrls.SearchLabels(), search.Parameters, AcceptHeaders.SearchCommitsPreview);
```

File: Octokit/Clients/SearchClient.cs [hunk 2]
```diff
@@ -83,5 +83,18 @@ public Task<SearchLabelsResult> SearchLabels(SearchLabelsRequest search)
...
         }
+
+        /// <summary>
+        /// search commits
+        /// https://developer.github.com/v3/search/#search-commits
+        /// </summary>
+        /// <param name="search"></param>
+        /// <returns>List of commits</returns>
+        [ManualRoute("GET", "/search/commits")]
+        public Task<SearchCommitsResult> SearchCommits(SearchLabelsRequest search)
```
Comment (shiftkey) [inline Octokit/Clients/SearchClient.cs]:
```suggestion
        public Task<SearchCommitsResult> SearchCommits(SearchCommitsRequest search)
```

File: Octokit/Models/Response/SearchCommitsResult.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections.Generic;
+using System.Diagnostics;
+using Octokit.Internal;
+
+namespace Octokit
+{
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    public class SearchCommitsResult : SearchResult<SearchCode>
```
Comment (shiftkey) [inline Octokit/Models/Response/SearchCommitsResult.cs]: You'll need a `SearchCommit` result that matches the shape from here:

https://developer.github.com/v3/search/#example-1

File: Octokit/Clients/SearchClient.cs (hunk 1 above)
Comment (shiftkey) [reply Octokit/Clients/SearchClient.cs]: `ApiUrls.SearchCommits` doesn't exist yet, but should be `search/commits`

**Comment by kentahikaru** [comment]
> Thanks for starting this off @kentahikaru!
> 
> I added some suggestions to help flesh out the important parts - there's probably more we'll uncover as we go...

**Comment by kentahikaru** [comment]
Hello @shiftkey, thanks for the suggestions.

File: Octokit/Models/Request/SearchCommitsRequest.cs [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+            get { return SortField.ToParameter(); }
+        }
+
+        /// <summary>
+        /// Finds commits authored by specified name
+        /// </summary>
+        /// <remarks>
+        /// https://help.github.com/en/github/searching-for-information-on-github/searching-commits#search-by-author-or-committer
+        /// </remarks>
+        public string Author;
```
Comment (shiftkey) [inline Octokit/Models/Request/SearchCommitsRequest.cs:55]: We have a convention in this codebase to ensure response models are properties defined with `{ get; protected set; }`. Are you able to add these?

File: Octokit/Helpers/AcceptHeaders.cs [hunk 5]
```diff
@@ -74,6 +74,8 @@ public static class AcceptHeaders
 
         public const string OAuthApplicationsPreview = "application/vnd.github.doctor-strange-preview+json";
 
+        public const string SearchCommitsPreview = "application/vnd.github.cloak-preview";
```
Comment (shiftkey) [inline Octokit/Helpers/AcceptHeaders.cs:77]: To get the build passing you will need to this rule to the `previewKeys` array in `Octokit.Tests.Conventions/PreviewsTests.cs`

https://github.com/octokit/octokit.net/blob/e41b2d19d74d84e5b5222ebce58d64fc308b1c2e/Octokit.Tests.Conventions/PreviewsTests.cs#L11-L21

File: build/.vscode/launch.json [hunk 6]
```diff
@@ -1,28 +0,0 @@
-{
```
Comment (shiftkey) [inline build/.vscode/launch.json:1]: I'm not sure why these files have been deleted, but it'd be great to keep them around.

**Pull request closed without merging**